drgn_tools/dm.py

```diff
@@ -258,9 +258,38 @@ def show_table_multipath(dm: Object, name: str) -> None:
     print(line)
 
 
+def show_table_snapshot(dm: Object, name: str) -> None:
+    msg = ensure_debuginfo(dm.prog_, ["dm_snapshot"])
+    if msg:
+        print(msg)
+        return
+
+    table = dm_table(dm)
+    for tid in range(table.num_targets):
+        target = table.targets[tid]
+        snapshot = cast("struct dm_snapshot *", target.private)
+        print(
+            "%s: %d %d %s %s %s %s %s %s %d"
```

Comment:
I'm not terribly familiar, does this need a table header? If so, it would be good to use the helpers from `drgn_tools/table.py` to print a header. As it is, I'm not sure if I'd know what the fields represent. But maybe this is similar to an existing format the kernel already outputs? Up to you whether you think this output is suitable.

Reply:
Unfortunately there isn't a fixed table header for this, different target driver type have different output, the first 4 columns will be in the same format, the following will be dynamic depending on the target driver type in the 4th column. And yes this is to simulate the the cmd output of `dmsetup table` .

Reply:
Ok, sounds good then.